src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -609,18 +609,27 @@ public ContainerExit waitContainer(final String containerId)
   @Override
   public void removeContainer(final String containerId)
       throws DockerException, InterruptedException {
-    removeContainer(containerId, false);
+    removeContainer(containerId, new RemoveContainerParam[0]);
   }
 
+  @Deprecated
   @Override
   public void removeContainer(final String containerId, final boolean removeVolumes)
       throws DockerException, InterruptedException {
+    removeContainer(containerId, RemoveContainerParam.removeVolumes());
+  }
+
+  @Override
+  public void removeContainer(final String containerId, final RemoveContainerParam... params)
+      throws DockerException, InterruptedException {
     try {
-      final WebTarget resource = resource()
-          .path("containers").path(containerId);
-      request(DELETE, resource, resource
-          .queryParam("v", String.valueOf(removeVolumes))
-          .request(APPLICATION_JSON_TYPE));
+      WebTarget resource = resource().path("containers").path(containerId);
+
+      for (final RemoveContainerParam param : params) {
+        resource = resource.queryParam(param.name().toLowerCase(Locale.ROOT), param.value());
```

Comment:
I think this pattern probably exists already elsewhere in the class, but instead of `.name().toLowerCase(locale)` why not just make sure that the enum `.name()` method returns the value to use in the query parameter? Or explicitly name the enum method `enum.queryParameterName()`

the locale bit is the part here that throws me off